packages/keyring-controller/src/KeyringController.ts

```diff
@@ -853,13 +887,16 @@ export class KeyringController extends BaseController<
   /**
    * Returns the public addresses of all accounts from every keyring.
    *
+   * @param keyringId - The id of the keyring to get the accounts from.
    * @returns A promise resolving to an array of addresses.
    */
-  async getAccounts(): Promise<string[]> {
-    return this.state.keyrings.reduce<string[]>(
-      (accounts, keyring) => accounts.concat(keyring.accounts),
-      [],
-    );
+  async getAccounts(keyringId?: string): Promise<string[]> {
+    return this.state.keyrings
+      .filter((keyring) => (keyringId ? keyring.id === keyringId : true))
+      .reduce<string[]>(
+        (accounts, keyring) => accounts.concat(keyring.accounts),
+        [],
+      );
```

Comment:
Clients could autonomously execute this operation by filtering `KeyringController`'s state already